client/memez/projects/memez/src/app/stores/entities/post.store.ts

```diff
@@ -1,69 +1,96 @@
-import {Injectable} from '@angular/core';
-import {action, observable} from 'mobx-angular';
-import {RootStore} from '../root.store';
-import {IPost} from '../../types/Entities/IPost';
+import {Injectable}                   from '@angular/core';
+import {action, computed, observable} from 'mobx-angular';
+import {RootStore}                    from '../root.store';
+import {IPost}                        from '../../../../../../../../shared/types/Entities/IPost';
+import {autorun}                      from 'mobx';
 
-import * as dayjs from 'dayjs';
-import {autorun} from 'mobx';
-import {MOCK_POSTS} from '../../../../../../../../shared/mock/MOCK_POSTS';
-import {ILike} from '../../types/Entities/ILike';
 
 @Injectable({
   providedIn: 'root'
 })
 export class PostStore {
   @observable posts: IPost[] = [];
-  useMock: boolean = true;
+
+  /*useMock: boolean = false;*/
 
   constructor(
     public root: RootStore
   ) {
     this.root.ps = this;  //self-registration at root store
     window['postStore'] = this;
+    autorun(() => {
+
+    })
+
+
+  }
+
+  @computed get reversedPosts() {
+    return this.posts
+      .slice()
+      .reverse()
+  }
 
+  @computed get userPosts() {
+    const userPosts: IPost[] = this.posts
+      .filter(post => post.postedBy._id === this.root.log.currentUser._id)
+    return userPosts.reverse()
   }
 
+  
+
+
+
   @action
   async getPosts(): Promise<IPost[]> {
-    if (this.useMock) {
-      return this.posts = MOCK_POSTS;
-    } else {
-      this.posts = await this.root
-        .postAdapter
-        .getPosts();
-    }
+    /* if (this.useMock) {
+     return this.posts = MOCK_POSTS;
+     } */
+
+    this.posts = await this.root
+      .postAdapter
+      .getPosts();
+
+   /* this.posts
+      .map(async post => {
+        await this.root
+          .likeStore
+          .getPostLikes(post)
+      })*/
+
+    return this.posts;
+
   }
```

Comment:
this function is a little bit confusing, because why does it need to assign the server response to this.posts and also return it? not  clear. 

Usually these types of functions do either not both